edx_solutions_api_integration/courses/views.py

```diff
@@ -1329,7 +1330,7 @@ def get_queryset(self):
         )
         users = users.select_related('profile')
         users = users.prefetch_related('user_attributes')
-        return users
+        return users.order_by(ordering)
```

Comment:
We could make this safer (e.g. `return users` on `except`), because it will fail with `FieldError` when the `ordering` is not valid field of the `User` model.

Reply:
Please also add this query param to the [docstring](https://github.com/edx-solutions/api-integration/pull/178/files#diff-ad7214ce503e395aadc244e70a926684R1189).

Reply:
:+1:

Reply:
Actually here we're just getting the QuerySet, but not resolving it by calling the DB, and FieldError is only raised when going further down and trying to resolve it. We also can't just return a 400 error response here, because it wouldn't propagate to the user - this function isn't a view.

I'll look at some other way of handling it, but I'm thinking it may be OK to throw a 500 here, since this is a server-to-server API and I kind of doubt we'd pass unchecked user input into this; a dev would notice the 500 while developing.

Reply:
I'll just check the field with `get_field`.

Reply:
I guess something like this should work fine:

```python
try:
    User._meta.get_field(ordering)
    return users.order_by(ordering)
except FieldDoesNotExist:
    return users
```

> I'll look at some other way of handling it, but I'm thinking it may be OK to throw a 500 here, since this is a server-to-server API and I kind of doubt we'd pass unchecked user input into this; a dev would notice the 500 while developing.

I had to:
![...](https://i.imgur.com/uZRoAso.png)

Reply:
Hah, yeah that seems right :p I ended up doing something similar.

Reply:
Ugh, I forgot to catch the error.

Reply:
I'm not sure if this will work:
https://docs.djangoproject.com/en/1.8/ref/models/meta/#django.db.models.options.Options.get_field

It raises `FieldDoesNotExist`.